spec/system/page_objects/modals/sidebar_edit_tags.rb

```diff
@@ -6,12 +6,10 @@ module PageObjects
   module Modals
     class SidebarEditTags < SidebarEditNavigationModal
       def has_tag_checkboxes?(tags)
-        tag_names = tags.map(&:name)
+        tag_checkboxes =
+          all(".sidebar-tags-form .sidebar-tags-form__tag-label-name", count: tags.length)
 
-        has_css?(".sidebar-tags-form .sidebar-tags-form__tag", count: tag_names.length) &&
-          all(".sidebar-tags-form .sidebar-tags-form__tag").all? do |row|
-            tag_names.include?(row["data-tag-name"].to_s)
-          end
+        expect(tag_checkboxes.map(&:text)).to eq(tags.map(&:name))
```

Comment:
If you don't care about ordering probably better to do `match_array`, otherwise this is fine :+1:

Reply:
The ordering matters here so we have to do an exact match.